ComposeGallery/src/androidTest/java/com/microsoft/device/display/samples/composegallery/PaneSynchronizationTest.kt

```diff
@@ -0,0 +1,158 @@
+        check(viewModel.imageSelectionLiveData.value == 7) { "Expected: 7 Actual: ${viewModel.imageSelectionLiveData.value}" }
+
+        // Check that detail pane has updated with the correct information
+        composeTestRule.onNode(
+            hasText(lastItem.id)
+                and hasAnySibling(hasContentDescription(lastItem.contentDescription))
+        ).assertIsDisplayed()
+        composeTestRule.onNode(
+            hasContentDescription(lastItem.contentDescription)
+                and hasAnySibling(hasText(lastItem.id))
+        ).assertIsDisplayed()
+
+        // REVISIT: added to make tests consistent, can remove when state isn't saved between tests
+        // Close the app
+        device.closeStart()
+    }
+
+    /**
+     * Test that a selection made when unspanned is remembered when span state changes
+     */
+    @Test
+    fun app_testSelectionPersistenceAfterSpan() {
+        val viewModel =
+            ViewModelProvider(composeTestRule.activity).get(AppStateViewModel::class.java)
+        val windowLayoutInfo = composeTestRule.activity.windowInfoRepository().windowLayoutInfo
+
+        composeTestRule.setContent {
+            ComposeGalleryTheme {
+                ComposeGalleryApp(viewModel = viewModel, windowLayoutInfo = windowLayoutInfo)
+            }
+        }
+
+        // Click on third surface duo entry
+        composeTestRule.onNodeWithContentDescription(getString(R.string.duo3_content_des))
+            .performClick()
+
+        // Check that detail view of third surface duo is displayed
+        composeTestRule.onNodeWithText(getString(R.string.duo3_id)).assertIsDisplayed()
+
+        // Check that viewmodel contains correct selection index
+        check(viewModel.imageSelectionLiveData.value == 2) { "Expected: 2 Actual: ${viewModel.imageSelectionLiveData.value}" }
+
+        // Span the app
+        if (USE_SWIPE_GESTURE)
+            device.spanFromStart()
+        publisherRule.simulateVerticalFold(composeTestRule)
+
+        // Check that third surface duo image is still displayed
+        composeTestRule.onNode(
+            hasContentDescription(getString(R.string.duo3_content_des))
+                and hasAnySibling(hasText(getString(R.string.duo3_id)))
+        ).assertIsDisplayed()
+
+        // Unspan the app
+        if (USE_SWIPE_GESTURE)
+            device.unspanToStart()
+        publisherRule.simulateNoFold()
+
+        // Check that detail view of third surface duo is still displayed
+        composeTestRule.onNodeWithText(getString(R.string.duo3_id)).assertIsDisplayed()
+
+        // REVISIT: added to make tests consistent, can remove when state isn't saved between tests
+        // Return to list view
+        composeTestRule.onNodeWithContentDescription(getString(R.string.switch_to_list))
+            .performClick()
+
+        // Close the app
+        device.closeStart()
+    }
+
+    private fun getString(@StringRes id: Int): String {
```

Comment:
Can we move it to the `Helper`?

Reply:
yes, good idea! I filed a feature request for the compose team as well, so hopefully in the future we are able to just use the string ID for the compose test matchers